addons/mail/static/src/utils/messaging_component.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module */
+
+import { useModels } from "@mail/component_hooks/use_models/use_models";
+import { useShouldUpdateBasedOnProps } from "@mail/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props";
+
+const componentRegistry = {};
+
+/**
+ * Allows a component to lean on the messaging features: the component is
+ * subscribed to changes in messaging models in its environment, the component
+ * will only be rerenderd if its props change, and the component has access to
+ * all components that are registered with this function without having to
+ * explicitly import them.
+ *
+ * @param {function} ComponentClass the constructor of the component to be
+ *      registered. Its name will be used as its key in the registry.
+ * @param {Object} [param1]
+ * @param {Object} [param1.propsCompareDepth] @see useShouldUpdateBasedOnProps
+ */
+export function registerMessagingComponent(ComponentClass, { propsCompareDepth = {} } = {}) {
+    ComponentClass.components = Object.assign(Object.create(componentRegistry), ComponentClass.components);
+    const originalSetup = ComponentClass.prototype.setup;
+    ComponentClass.prototype.setup = function setup(...args) {
+        originalSetup.apply(this, args);
+        // useModels must be defined after useRenderedValues, indeed records and
+        // fields accessed during useRenderedValues should be observed by
+        // useModels as if they were part of the OWL rendering itself.
+        useModels();
+        useShouldUpdateBasedOnProps({ propsCompareDepth });
+    }
+    const { name } = ComponentClass;
```

Comment:
As per discussed in another comment but I prefer to mention at the correct place.
It would be better if this was making a new class, instead of changing the given one in place.